core/arch/aarchxx/mangle.c

```diff
@@ -149,32 +153,31 @@ create_load_or_store_instr(dcontext_t *dcontext, reg_id_t reg, opnd_t mem, bool
  * is odd. Optionally takes reg_skip into account.
  */
 static void
-insert_save_or_restore_registers(dcontext_t *dcontext, instrlist_t *ilist, instr_t *instr,
-                                 bool *reg_skip, reg_id_t base_reg, reg_id_t first_reg,
-                                 bool save, int reg_type,
-                                 opnd_t (*get_mem_opnd)(uint base_reg, bool is_single_reg,
-                                                        int reg_type, uint num_saved,
-                                                        callee_info_t *ci),
-                                 callee_info_t *ci)
+insert_save_or_restore_gpr_simd_registers(
+    dcontext_t *dcontext, instrlist_t *ilist, instr_t *instr, bool *reg_skip,
+    reg_id_t base_reg, reg_id_t first_reg, bool save, reg_type rtype,
+    opnd_t (*get_mem_opnd)(uint base_reg, bool is_single_reg, reg_type rtype,
+                           uint num_saved, callee_info_t *ci),
+    callee_info_t *ci)
 {
-    /* TODO i#5365: Add save/restore of SVE predicate registers. */
-    uint i, reg1 = UINT_MAX, num_regs = (reg_type == GPR_REG_TYPE) ? 30 : 32;
+    ASSERT(rtype == GPR_REG_TYPE || rtype == SIMD_REG_TYPE);
+
+    uint i, reg1 = UINT_MAX,
+            num_regs = (rtype == GPR_REG_TYPE) ? 30 : MCXT_NUM_SIMD_SLOTS;
     uint saved_regs = 0;
     instr_t *new_instr;
     /* Use stp/ldp to save/restore as many register pairs to memory, skipping
      * registers according to reg_skip.
      */
     for (i = 0; i < num_regs; i += 1) {
-        if (reg_type == SVE_REG_TYPE)
-            break;
         if (reg_skip != NULL && reg_skip[i])
             continue;
 
         if (reg1 == UINT_MAX)
             reg1 = i;
         else {
             opnd_t mem1 =
-                get_mem_opnd(base_reg, false /* is_single_reg */, reg_type,
+                get_mem_opnd(base_reg, false /* is_single_reg */, rtype,
```

Comment:
The convention is to use the parameter name with = in front: `/*is_single_reg=*/false`.  Ditto below.

Reply:
Done.